File: spring_server/src/main/java/com/semothon/spring_server/crawling/entity/UserCrawlingRecommendation.java
package com.semothon.spring_server.crawling.entity;

import com.semothon.spring_server.user.entity.User;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
@ToString(of = {"userCrawlingRecId", "score", "activityScore"})
@Table(name = "user_crawling_recommendations",
        indexes = {
        },
        uniqueConstraints = {
                @UniqueConstraint(columnNames = {"user_id", "crawling_id"})
        }
)
public class UserCrawlingRecommendation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long userCrawlingRecId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "crawling_id", nullable = false)
    private Crawling crawling;

    @Column(nullable = false)
    private Double score;

    @Column(nullable = false)
    private Double activityScore;

    public void updateUser(User user){
        this.user = user;
        user.addUserCrawlingRecommendation(this);
    }

    public void updateCrawlingData(Crawling crawling){
        this.crawling = crawling;
        crawling.addUserCrawlingRecommendation(this);
    }

    public void updateActivityScore(Double score) {
        this.activityScore = score;
    }

}

File: fastapi_server/app/models/user_crawling_recommendation.py
from sqlalchemy import Column, Float, ForeignKey, Integer, UniqueConstraint
from sqlalchemy.orm import relationship
from database import Base

class UserCrawlingRecommendation(Base):
    __tablename__ = "user_crawling_recommendations"
    __table_args__ = (
        UniqueConstraint("user_id", "crawling_id", name="uq_user_crawling"),
    )

    user_crawling_rec_id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    user_id = Column(Integer, ForeignKey("users.user_id"), nullable=False)
    crawling_id = Column(Integer, ForeignKey("crawling.crawling_id"), nullable=False)
    score = Column(Float, nullable=False)
    activity_score = Column(Float, nullable=False)

    user = relationship("User")
    crawling = relationship("Crawling")
